optaplanner-core/src/main/java/org/optaplanner/core/impl/util/ConsecutiveData.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.impl.util;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class ConsecutiveData<T, D extends Comparable<D>> {
+    private final ConsecutiveSetTree<T, ?, D> sourceTree;
```

Comment:
Please pick descriptive names for `T`, `D`. We may have been able to get away with `A` and `B` in CS, but those are purely positional names - in this case, these types have meaning, and their names should carry it.